src/entities/trades/swapr-v3/SwaprV3.ts

```diff
@@ -0,0 +1,209 @@
+import type { BaseProvider } from '@ethersproject/providers'
+import dayjs from 'dayjs'
+
+import { Currency, Fraction, validateAndParseAddress } from '@uniswap/sdk-core'
+import invariant from 'tiny-invariant'
+
+import { CurrencyAmount, Percent, Price, TokenAmount } from '../../fractions'
+
+import { TradeWithSwapTransaction } from '../interfaces/trade'
+import { RoutablePlatform } from '../routable-platform'
+import { getProvider, tryGetChainId } from '../utils'
+import { ONE, TradeType } from '../../../constants'
+
+import { UnsignedTransaction } from 'ethers'
+import { TradeOptions } from '../interfaces/trade-options'
+import { parseUnits } from '@ethersproject/units'
+import { Route } from './entities/route'
+import { SWAPR_ALGEBRA_CONTRACTS } from './constants'
+import { getQuoterContract, getRouterContract } from './contracts'
+import { getRoutes } from './routes'
+
+interface SwaprV3ConstructorParams {
+  maximumSlippage: Percent
+  inputAmount: CurrencyAmount
+  outputAmount: CurrencyAmount
+  tradeType: TradeType
+  chainId: number
+  priceImpact: Percent
+  fee: Percent
+}
+
+export class SwaprV3Trade extends TradeWithSwapTransaction {
+  public constructor({
+    inputAmount,
+    outputAmount,
+    maximumSlippage,
+    priceImpact,
+    tradeType,
+    chainId,
+    fee,
+  }: SwaprV3ConstructorParams) {
+    super({
+      details: undefined,
+      type: tradeType,
+      inputAmount,
+      outputAmount,
+      maximumSlippage,
+      platform: RoutablePlatform.SWAPR_V3,
+      chainId,
+      executionPrice: new Price({
+        baseCurrency: inputAmount.currency,
+        quoteCurrency: outputAmount.currency,
+        denominator: inputAmount.raw,
+        numerator: outputAmount.raw,
+      }),
+      priceImpact,
+      fee,
+      approveAddress: SWAPR_ALGEBRA_CONTRACTS['router'],
+    })
+  }
+
+  static async getQuote(
+    { amount, quoteCurrency, tradeType, maximumSlippage }: any,
```

Comment:
Let's define a type here

Reply:
Having a type here is provoking a cascading of small type errors.

Reply:
Attempted to fix it, not perfect but better.